From 2cd03491223cdd21820061678c494e4dea0c5765 Mon Sep 17 00:00:00 2001
From: Yongqiang Sun <yongqiang.sun@amd.com>
Date: Thu, 4 Aug 2016 10:20:18 -0400
Subject: [PATCH 0723/1722] drm/amd/dal: optimized eDP dc_link_detect.

When resume doing dc_link_detect, in case of eDP,
there isn't necessary to disconnect local sink and
create a new sink, keep local sink unchanged will
save some detection time.

Signed-off-by: Yongqiang Sun <yongqiang.sun@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index fc43c5c..b2c53f6 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -577,6 +577,10 @@ bool dc_link_detect(const struct dc_link *dc_link, bool boot)
 		return false;
 	}
 
+	if (link->public.connector_signal == SIGNAL_TYPE_EDP &&
+			link->public.local_sink)
+		return true;
+
 	link_disconnect_sink(link);
 
 	if (new_connection_type != dc_connection_none) {
-- 
2.7.4

